homeassistant/components/aosmith/coordinator.py

```diff
@@ -0,0 +1,53 @@
+"""The data update coordinator for the A. O. Smith integration."""
+
+from asyncio import timeout
+import logging
+from typing import Any
+
+from py_aosmith import (
+    AOSmithAPIClient,
+    AOSmithInvalidCredentialsException,
+    AOSmithUnknownException,
+)
+
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DOMAIN, FAST_INTERVAL, REGULAR_INTERVAL
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class AOSmithCoordinator(DataUpdateCoordinator[dict[str, dict[str, Any]]]):
+    """Custom data update coordinator for A. O. Smith integration."""
+
+    def __init__(self, hass: HomeAssistant, client: AOSmithAPIClient) -> None:
+        """Initialize the coordinator."""
+        super().__init__(hass, _LOGGER, name=DOMAIN, update_interval=REGULAR_INTERVAL)
+        self.client = client
+
+    async def _async_update_data(self):
+        """Fetch latest data from API."""
+        try:
+            async with timeout(10):
```

Comment:
Shouldn't the timeout be relocated in the library?

Reply:
I don't think it makes sense to move the timeout into the library. There is nothing about this API that requires a specific timeout value to be used. Also, many integrations, such as google tasks, are using a timeout on their API call.

Reply:
After discussing on Discord I have moved the timeout into the library.